Tests/SonarScanner.MSBuild.Common.Test/EnvironmentBasedPlatformHelperTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static SonarScanner.MSBuild.Common.EnvironmentBasedPlatformHelper;
+
+namespace SonarScanner.MSBuild.Common.Test;
+
+[TestClass]
+public class EnvironmentBasedPlatformHelperTests
+{
+    [TestMethod]
+    public void GetFolderPath_WithUserProfile()
+    {
+        Instance.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None).Should().NotBeNull();
```

Comment:
You could add some assertions for platform-specific results:
```cs
if (Instance.IsWindows())
{
    userProfile.Should().Contain("AppData");
}
...
```

Reply:
I can't find anything that really is OS-dependent, but isn't machine-dependent.
The user profile won't contain `AppData`. Moreover:
- the drive can be any, and even not a single letter (you can also have UNC paths)
- `Users` has changed over time with windows versions (e.g. it used to be `Documents and Settings`)
- the name of the user can really be anything
- `System` can be `WINNT` instead of `Windows` (at least, [it used to](https://answers.microsoft.com/en-us/windows/forum/all/should-the-folder-cwinnt-be-present-on-all/e376a2a8-19cc-40d0-a63c-081b107798d1))
- `ProgramFiles` and `ProgramFilesX86` are CPU-dependent, and Windows on ARM would likely break the test
- `ApplicationData` can be moved around.
So I would avoid adding some hardcoded string and risk having a test with a flaky or context-dependent result.
@martin-strecker-sonarsource  Any idea?

Reply:
This is somewhat useful: Assert the returned path equals [`Environment.GetFolderPath`](https://learn.microsoft.com/en-us/dotnet/api/system.environment.getfolderpath?view=net-8.0#system-environment-getfolderpath(system-environment-specialfolder-system-environment-specialfolderoption)).

Reply:
Given that `EnvironmentBasedPlatformHelperTests` tests the `Environment`-based implementation of `IPlatformHelper`, we are basically testing that `Environment.GetFolderPath` is equal to itself... which would be even `false` in the very unlikely but technically possible even that the user profile is moved to another path during the call.

Because the event seems remote, I have made the change. The value brought by this change remains dubious to me: one thing is to assert that the method gives a non-null result. Another is to check that the method returns the same as its one-line invocation to `Environment`.